src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -228,7 +228,7 @@ private async Task ProcessQueueAsync()
                         // Non mutating are fire-and-forget because they are by definition readonly. Any errors
                         // will be sent back to the client but we can still capture errors in queue processing
                         // via NFW, though these errors don't put us into a bad state as far as the rest of the queue goes.
-                        _ = ExecuteCallbackAsync(work, context, _cancelSource.Token).ReportNonFatalErrorAsync();
+                        _ = Task.Run(() => ExecuteCallbackAsync(work, context, _cancelSource.Token).ReportNonFatalErrorAsync(), _cancelSource.Token);
```

Comment:
Might be helpful to leave a little comment about why Task.Run is not equivalent to just calling the method in question.

Reply:
Sure can totally add that!